### PR TITLE
Add function `get_chr_x_hom_alt_cutoffs`, add arguments to `infer_sex_karyotype` and `get_sex_expr` to use the new function and it's output.

### PR DESCRIPTION
**Pull request by jkgoodrich**
Function `get_chr_x_hom_alt_cutoffs` gets cutoffs for the fraction homozygous alternate genotypes on chromosome X in 'XY' and 'XX' samples.
Changes to `infer_sex_karyotype` and `get_sex_expr` are to use this new function to determine  cutoffs for the fraction of homozygous alternate genotypes (hom-alt/(hom-alt + het)) on chromosome X for for 'XX' and 'XY' karyotypes and then use them to determine final sex karyotypes.

### CODE REVIEW

File: gnomad/sample_qc/pipeline.py [hunk 1]
```diff
@@ -291,16 +305,40 @@ def infer_sex_karyotype(
...
+        chr_x_frac_hom_alt_cutoffs = get_chr_x_hom_alt_cutoffs(
+            ploidy_ht,
+            chr_x_frac_hom_alt_expr,
+            f_stat_cutoff=f_stat_cutoff,
+            group_by_expr=group_by_expr,
+            cutoff_stdev=normal_chr_x_hom_alt_cutoff,
+        )
+
+    else:
+        chr_x_frac_hom_alt_expr = None
```
Comment (klaricch) [inline gnomad/sample_qc/pipeline.py:325]: I don't think you need this since it already defaults to None?

**Review by klaricch — CHANGES_REQUESTED** [review]

File: gnomad/sample_qc/pipeline.py (hunk 1 above)
Comment (jkgoodrich) [reply gnomad/sample_qc/pipeline.py:325]: oops, yup, thank you. I think I reorganized things and accidentally left it.

File: gnomad/sample_qc/pipeline.py [hunk 2]
```diff
@@ -252,6 +253,8 @@ def infer_sex_karyotype(
     use_gaussian_mixture_model: bool = False,
     normal_ploidy_cutoff: int = 5,
     aneuploidy_cutoff: int = 6,
+    chr_x_frac_hom_alt_expr: Optional[hl.expr.NumericExpression] = None,
+    normal_chr_x_hom_alt_cutoff: int = 5,
```
[the remark this answers is not in the document]
Comment (jkgoodrich) [reply gnomad/sample_qc/pipeline.py:257]: I think I would rather keep a default so we don't need to always supply one. Would it help to add to the logger below to print out the stdev that is being used?

[the remark this answers is not in the document]
Comment (klaricch) [reply gnomad/sample_qc/pipeline.py:257]: Sounds good. Yeah I like adding it to the logger.

**Review by klaricch — APPROVED** [review]
lgtm!

**Pull request merged**